tools/benchmarksAnalyzer/src/main/kotlin/org/jetbrains/analyzer/SummaryBenchmarksReport.kt

```diff
@@ -0,0 +1,204 @@
+    private val benchmarksWithChangedStatus = mutableListOf<FieldChange<BenchmarkResult.Status>>()
+
+    // Maps with changes of performance.
+    var regressions = mapOf<String, ScoreChange>()
+        private set
+    var improvements = mapOf<String, ScoreChange>()
+        private set
+
+    // Summary value of report - geometric mean.
+    lateinit var geoMeanBenchmark: SummaryBenchmark
+        private set
+    var geoMeanScoreChange: ScoreChange? = null
+        private set
+
+    // Changes in environment and tools.
+    var envChanges = listOf<FieldChange<String>>()
+        private set
+    var kotlinChanges = listOf<FieldChange<String>>()
+        private set
+
+    // Countable properties.
+    val failedBenchmarks: List<String>
+        get() = mergedReport.filter { it.value.first?.meanBenchmark?.status == BenchmarkResult.Status.FAILED }
+                            .map { it.key }
+
+    val addedBenchmarks: List<String>
+        get() = mergedReport.filter { it.value.second == null }.map { it.key }
+
+    val removedBenchmarks: List<String>
+        get() = mergedReport.filter { it.value.first == null }.map { it.key }
+
+    val benchmarksNumber: Int
+        get() = mergedReport.keys.size
+
+    val currentMeanVarianceBenchmarks: List<MeanVarianceBenchmark>
+        get() = mergedReport.filter { it.value.first != null }.map { it.value.first!! }
+
+    init {
+        // Count avarage values for each benchmark.
+        val currentBenchmarksTable = collectMeanResults(currentReport.benchmarks)
+        val previousBenchmarksTable = previousReport?.let {
+            // Check changes in environment and tools.
+            analyzeEnvChanges(currentReport.env, previousReport.env)
+            analyzeKotlinChanges(currentReport.compiler, previousReport.compiler)
+            collectMeanResults(previousReport.benchmarks)
+        }
+        createMergedReport(currentBenchmarksTable, previousBenchmarksTable)
+        previousBenchmarksTable?.let {
+            analyzePerformanceChanges()
+        }
+    }
+
+    fun getBenchmarksWithChangedStatus(): List<FieldChange<BenchmarkResult.Status>> = benchmarksWithChangedStatus
+
+    // Create geometric mean.
+    private fun createGeoMeanBenchmark(benchTable: BenchmarksTable): MeanVarianceBenchmark {
+        val geoMeanBenchmarkName = "Geometric mean"
+        val geoMean = geometricMean(benchTable.toList().map { (_, value) -> value.meanBenchmark.score })
+        val varianceGeoMean = geometricMean(benchTable.toList().map { (_, value) -> value.varianceBenchmark.score })
+        val meanBenchmark = BenchmarkResult(geoMeanBenchmarkName, geoMean)
+        val varianceBenchmark = BenchmarkResult(geoMeanBenchmarkName, varianceGeoMean)
+        return MeanVarianceBenchmark(meanBenchmark, varianceBenchmark)
+    }
+
+    // Merge current and compare to report.
+    private fun createMergedReport(currentBenchmarks: BenchmarksTable, previousBenchmarks: BenchmarksTable?) {
```

Comment:
You can split this method into `m1` and `m2` returning merged bucket and geomean, respectively. Then in `init` block 
```
mergedReport = m1(..)
geoMeanBenchmark = m2(...)
```
Which will allow to mark these variables as `val` as it should be.